### DISCUSSION TITLE
Article: Text Alignment Best Practises

### DISCUSSION THREAD

**Discussion by mrosvik** [digdir/designsystemet#5290] [Best practices]
There have been some questions about text alignment. We should add an article under "[God praksis](https://www.designsystemet.no/god-praksis)" about this. 

UU-tilsynet says nothing about this:
https://www.uutilsynet.no/veiledning/tekst-og-struktur/226
Relevant WCAG Criterion:
> Width is no more than 80 characters or glyphs (40 if CJK).
Text is not justified (aligned to both the left and the right margins).

[WCAG: 1.4.8 Visual presentation (AAA)](https://www.w3.org/WAI/GL/UNDERSTANDING-WCAG20/visual-audio-contrast-visual-presentation.html)

Inspiration: 
- https://blog.prototypr.io/text-alignment-best-practises-c4114daf1a9b
> "When text is longer than two or three lines, it is always recommended to ignore center alignment."

- https://thewebsitearchitect.com/does-center-aligned-text-matter-for-accessibility/
> Left is the most popular and default text alignment. Its the best for readability and user experience because of the way our eyes read.

- [Article on justified text](https://www.boia.org/blog/why-justified-or-centered-text-is-bad-for-accessibility)
- [W3 supplemental guidance on justified text](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Supplemental_Guidance:_Text_Justification)

- Some small mentions in gov.uk 
https://accessibility.blog.gov.uk/2016/09/02/dos-and-donts-on-designing-for-accessibility/
https://design-system.service.gov.uk/styles/typography/#text-alignment-override-classes
- Other sources?

Question:

![Image](https://github.com/digdir/designsystem/assets/1464915/1a8c230e-c909-4c84-8fbe-b7992a90a4da)

```[tasklist]
### Tasks
- [x] Finne stoff og kvalitetssikre
- [x] Skrive utkast
- [x] Sende til Dorte for kvalitetssikring
- [x] Justere etter tilbakemelding
- [x] Dele med Marianne
- [ ] Marianne gir meg beskjed om eventuelle endringer og setter til  Done
```

**Comment by Ildest** [comment]
Jeg ser på eksempelet over. Med utgangspunkt i artikkelen du delte @mrosvik, så kan vi sentrere her fordi teksten er kort, men vi trenger å dele kunnskap om å skrive mer effektive og kortere tekster ser jeg. He kan teksten egentlig bare være: Legg inn koden fra brevet du fikk i posten. Det er også noen anglisismer som sniker seg inn i norsken vår: På norsk fyller vi "ut" et skjema, mens på engelsk sier de "fill in". Da får vi ofte denne overføringsfeilen med "fyll inn". Da er det bedre å si legg inn her hvor det er kun et felt.

**Comment by Ildest** [comment]
Artikkelen er ferdig og delt med @mrosvik så hun kan gjennomgå den og legge den i Storybook.
